assets/src/edit-story/components/form/numeric.js

```diff
@@ -97,48 +111,124 @@ function Numeric({
   canBeEmpty,
   ...rest
 }) {
-  const isMultiple = value === MULTIPLE_VALUE;
-  const placeholder = isMultiple ? __('multiple', 'web-stories') : '';
-  const [dot, setDot] = useState(false);
   const [inputValue, setInputValue] = useState(value);
-  const ref = useRef();
 
-  useEffect(() => setInputValue(value), [value]);
+  const inputRef = useRef();
+  const targetFormRef = useRef(null);
+  const skipValidationRef = useRef(false);
+
+  const { focused, handleFocus, handleBlur } = useFocusAndSelect(inputRef);
+
+  const isMultiple = inputValue === MULTIPLE_VALUE;
+  const placeholder = isMultiple ? __('multiple', 'web-stories') : '';
+
+  const handleChange = useCallback((event) => {
+    setInputValue(event.target.value);
+  }, []);
+
+  const validateAndSubmitInput = useCallback(() => {
+    targetFormRef.current = inputRef.current.form;
+
+    const validValue = validateInput(inputValue, {
+      canBeNegative,
+      canBeEmpty,
+      float,
+    });
+
+    setInputValue(validValue ?? value);
+    onChange(validValue ?? value);
+  }, [inputValue, canBeNegative, canBeEmpty, float, value, onChange]);
 
   const handleUpDown = useCallback(
     ({ key, altKey }) => {
       if (isMultiple) {
         return;
       }
 
-      let newValue;
+      const validValue = validateInput(inputValue, {
+        canBeNegative,
+        canBeEmpty,
+        float,
+      });
```

Comment:
This (including `?? value`) is currently repeated a few times, could be moved to a common function.